src/main/java/org/icgc/argo/program_service/services/InvitationService.java

```diff
@@ -39,17 +49,23 @@ public UUID inviteUser(@NotNull ProgramEntity program,
     return invitation.getId();
   }
 
-  public boolean acceptInvite(UUID invitationId) {
-    val invitation = invitationRepository.findById(invitationId);
-    if (invitation.isPresent()) {
-      val i = invitation.get();
-      i.accept();
-      val email = invitation.get().getUserEmail();
-      egoService.joinProgram(email, i.getProgram().getShortName(), i.getRole());
-      invitationRepository.save(invitation.get());
-      return true;
-    } else {
-      return false;
-    }
+  public EgoUser acceptInvite(UUID invitationId) throws NotFoundException {
+    val invitation = invitationRepository
+            .findById(invitationId)
+            .orElseThrow(() ->
+              new NotFoundException(format("Cannot find invitation with id '%s' ", invitationId)));
+    invitation.accept();
+    egoService.joinProgram(invitation.getUserEmail(), invitation.getProgram().getShortName(), invitation.getRole());
+    invitationRepository.save(invitation);
```

Comment:
I would suggest making the `egoService` call the last call in this method. If you add `@Transactional` to this method, then if the `save` fails (i.e throws an exeception), a corrupting call was not made and db changes in the invitation repo are rolled back.